addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -58,8 +68,49 @@ def _process_step_create_lead(self, discuss_channel):
                 'partner_id': partner.id,
                 'company_id': partner.company_id.id,
             }
-
         create_values.update(self._chatbot_crm_prepare_lead_values(
             discuss_channel, customer_values['description']))
+        return self.env["crm.lead"].create(create_values)
 
-        self.env['crm.lead'].create(create_values)
+    def _process_step_create_lead_and_forward(self, discuss_channel):
+        lead = self._process_step_create_lead(discuss_channel)
+        teams = lead.team_id
+        if not teams:
+            possible_teams = self.env["crm.team"].search(
```

Comment:
Note that we could extract the domain into a helper used notably in '_cron_assign_leads' . Purpose is to highlight the fact that code should be aligned, hence adding some common nodes when possible (and making sense).